### PR TITLE
Feat: support tenant label in `eks-update-kubeconfig`

### PR DESCRIPTION
**Pull request by korenyoni**
## what

* Support optional tenant label in `eks-update-kubeconfig`

## why

* `null-label` supports an optional `tenant` label, however `eks-update-kubeconfig` does not.
* Misc: add additional help documentation explaining how to have a single kubeconfig file with multiple contexts, rather than a separate file for each context.

## references

- https://github.com/cloudposse/terraform-null-label/releases/tag/0.25.0
- requires https://github.com/cloudposse/atmos/pull/71
- https://github.com/cloudposse/atmos/releases/tag/v1.3.4

## tests

**NOTE**: outdated, see comments below

### with tenant label

<details>
<summary>eks-update-kubeconfig mgmt-uw2-sandbox</summary>

```bash
Updated context arn:aws:eks:us-west-2:[REDACTED]:cluster/[REDACTED]-mgmt-uw2-sandbox-eks-cluster in /conf/.kube/kubecfg.mgmt-uw2-sandbox-admin
```
</details>

### without tenant label

<details>
<summary>eks-update-kubeconfig ue1-sandbox</summary>

```bash
Added new context arn:aws:eks:us-east-1:[REDACTED]:cluster/[REDACTED]-ue1-sandbox-eks-cluster to /conf/.kube/kubecfg.ue1-sandbox-admin
```
</details>

### CODE REVIEW

**Review by nitrocode — APPROVED** [review]

**Comment by korenyoni** [comment]
I need to reformat — some tabs / spaces inconsistencies in indentation

File: rootfs/usr/local/bin/eks-update-kubeconfig [hunk 1]
```diff
@@ -25,16 +26,28 @@ Usage:
 
   With "off", deletes the currently active kubecfg file.
 
+  If it is desired to have single kubeconfig file with all contexts, as opposed to a separate kubeconfig file for each
```
Comment (osterman) [inline rootfs/usr/local/bin/eks-update-kubeconfig:29]:
```suggestion
  If it is desired to have a single kubeconfig file with all contexts, as opposed to a separate kubeconfig file for each
```

File: rootfs/usr/local/bin/eks-update-kubeconfig [hunk 2]
```diff
@@ -25,16 +26,28 @@ Usage:
 
   With "off", deletes the currently active kubecfg file.
 
+  If it is desired to have a single kubeconfig file with all contexts, as opposed to a separate kubeconfig file for each
+  context, override the EKS_KUBECONFIG_PATTERN environment variable such that it points to a single file, for example
+  /config/.kube/config. Afterwards, a tool such as kubectx can be used to switch between contexts.
+
```
Comment (Nuru) [inline rootfs/usr/local/bin/eks-update-kubeconfig:32]:
```suggestion
  If you chose to do this, remember that "off" deletes the entire kubeconfig file.
  
```

File: rootfs/usr/local/bin/eks-update-kubeconfig [hunk 3]
```diff
@@ -96,6 +113,9 @@ main() {
 		file_name $2 $role
 		return 0
 	fi
+	if [[ $1 =~ ^[[:alnum:]]+-[[:alnum:]]+-[[:alnum:]]+$ ]]; then
```
Comment (Nuru) [inline rootfs/usr/local/bin/eks-update-kubeconfig:116]: Although this is probably OK, I prefer the more explicit statement of our convention:
```suggestion
	if [[ $1 =~ ^[^-]+-[^-]+-[^-]+$ ]]; then
```

**Review by Nuru — CHANGES_REQUESTED** [review]

**Comment by korenyoni** [comment]
Tested this again in both an environment without a `tenant` label and one with.

**Comment by korenyoni** [comment]
Updated and tested manually:

<details>
<summary>with atmos < 1.3.4 in environment without the tenant label</summary>

```bash
$ eks-update-kubeconfig ue1-sandbox
WARNING:
The current version of atmos is 0.21.3 (a32a4a4), and the minimum version required to determine if the 'tenant' label
is being used is v1.3.4. Because of this, it is assumed 'tenant' is not used. If you would like to use the 'tenant' label,
update atmos to v1.3.4 or above.

Updated context arn:aws:eks:us-east-1:[REDACTED]:cluster/eg-ue1-sandbox-eks-cluster in /conf/.kube/kubecfg.ue1-sandbox-admin
```

</details>

<details>
<summary>with atmos >= 1.3.4 in environment without the tenant label</summary>

```bash
$ eks-update-kubeconfig ue1-sandbox
Updated context arn:aws:eks:us-east-1:[REDACTED]:cluster/eg-ue1-sandbox-eks-cluster in /conf/.kube/kubecfg.ue1-sandbox-admin
```

</details>

<details>
<summary>with atmos >= 1.3.4 in environment with the tenant label</summary>

```bash
$ eks-update-kubeconfig mgmt-uw2-sandbox
Updated context arn:aws:eks:us-west-2:[REDACTED]:cluster/eg-mgmt-uw2-sandbox-eks-cluster in /conf/.kube/kubecfg.mgmt-uw2-sandbox-admin
```

</details>

File: rootfs/usr/local/bin/eks-update-kubeconfig [hunk 4]
```diff
@@ -10,6 +10,7 @@
 # We expect AWS_DEFAULT_SHORT_REGION was set in Dockerfile, and if not there,
 # then in aws.sh, but just in case...
 AWS_DEFAULT_SHORT_REGION=${AWS_DEFAULT_SHORT_REGION:-$(aws-region --${AWS_REGION_ABBREVIATION_TYPE} ${AWS_DEFAULT_REGION:-us-west-2})}
+TENANT_LABEL_ENABLED=false
```
Comment (Nuru) [inline rootfs/usr/local/bin/eks-update-kubeconfig:13]: This raises a larger issue of how we support tenants in Geodesic (on the command line/console) versus how we support them via Spacelift. Do we add an environment variable in Dockerfile like we have for `NAMESPACE` and do we add tools for switching tenants along the lines of `assume-role`? We need to have a design sprint for this. @osterman @korenyoni @aknysh @nitrocode

**Review by Nuru — CHANGES_REQUESTED** [review]
Not requesting specific changes, but want to wait for a larger design conversation of which this is just one piece.

**Pull request closed without merging**